### PR TITLE
Adds Docker Build.

### PR DESCRIPTION
**Pull request by mhoak**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -47,30 +91,8 @@ jobs:
         id: plan
         env:
           CONTAINER: api_docker_image=gcr.io/google-samples/hello-app:2.0
-        run: terraform plan -out=plan.tfplan -var="api_docker_image=${CONTAINER}" $GITHUB_WORKSPACE/tf/
+        run: terraform plan -out=plan.tfplan -var="api_docker_image=us.gcr.io/otrego-dev/apiserver:${GITHUB_RUN_NUMBER}" $GITHUB_WORKSPACE/tf/
```
Comment (artasparks) [inline .github/workflows/go.yml:94]: what does GITHUB_RUN_NUMBER mean / correspond to?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -8,11 +8,11 @@ on:
...
 jobs:
 
-  build:
-    name: Build
+  test:
+    name: Unit Test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@master
```
Comment (artasparks) [inline .github/workflows/go.yml]: Is there not a recent release that we can use?

File: .github/workflows/go.yml (hunk 1 above)
Comment (mhoak) [reply .github/workflows/go.yml:94]: "A unique number for each run of a particular workflow in a repository. This number begins at 1 for the workflow's first run, and increments with each new run. This number does not change if you re-run the workflow run."


https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables

File: .github/workflows/go.yml (hunk 2 above)
Comment (mhoak) [reply .github/workflows/go.yml]: fixed

**Review by artasparks — APPROVED** [review]

**Pull request merged**